pkg/resource/stack/deployment.go

```diff
@@ -537,13 +522,13 @@ func collectCiphertexts(ciphertexts *[]string, prop interface{}) {
 }
 
 // DeserializeResource turns a serialized resource back into its usual form.
-func DeserializeResource(res apitype.ResourceV3, dec config.Decrypter, enc config.Encrypter) (*resource.State, error) {
+func DeserializeResource(res apitype.ResourceV3, dec config.Decrypter) (*resource.State, error) {
```

Comment:
The naming here is a bit confusing now.  We're getting only a decrypter passed in, but we're using it to encrypt later.  I get that we can get the same underlying type from go, but maybe we could name this a bit better?

Reply:
Yeah, this is a little bit of a hack still that, if the decrypter happens to be an encrypter, then we do the extra encrypt step.